lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb

```diff
@@ -116,12 +138,19 @@ def with_message(message)
           self
         end
 
+        def allow_nil(allow_nil = true)
+          @allow_nil = allow_nil
+          self
+        end
+
         def matches?(subject)
           super(subject)
           @expected_message ||= :blank
 
           if secure_password_being_validated?
             disallows_and_double_checks_value_of!(blank_value, @expected_message)
+          elsif @allow_nil
+            allows_value_of(nil)
```

Comment:
Okay, so this raises a question... If your `belongs_to` can be nil, but it can't be an empty string... then it can't ever be blank, can it? What are you trying to protect against?